### PR TITLE
highs: update to 1.5.1; remove redundant dependencies

### PR DESCRIPTION
**Pull request by galabovaa**
- [ x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

I cleared most audit warnings, there are two left but I don't know what to do about them. I see many other formulas would get them as well. They are
* 1: col 1: No Sorbet sigil found in file 
* 1: col 1: Missing frozen string literal comment.
-----

### CODE REVIEW

**Comment by galabovaa** [comment]
Closing, WIP

**Comment by galabovaa** [comment]
Please review and let me know if there are any issues! 

Thank you,
Ivet.  - from the HiGHS team

File: Formula/highs.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
+# Class for the HiGHS solver
```
Comment (ZhongRuoyu) [inline Formula/highs.rb:1]:
```suggestion
```

Please remove this comment.

**Review by ZhongRuoyu — APPROVED** [review]
Thanks, @galabovaa! Please use the [preferred commit message format](https://docs.brew.sh/Formula-Cookbook#commit) next time.

**Pull request closed without merging**